components/org.wso2.carbon.identity.auth.abstractotp/src/main/java/org/wso2/carbon/identity/auth/abstractotp/internal/AuthenticatorServiceComponent.java

```diff
@@ -0,0 +1,127 @@
+package org.wso2.carbon.identity.auth.abstractotp.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.event.services.IdentityEventService;
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.handler.event.account.lock.service.AccountLockService;
+import org.wso2.carbon.idp.mgt.IdpManager;
+import org.wso2.carbon.user.core.service.RealmService;
+
+/**
+ * OTP service component.
+ */
+@Component(
+        name = "carbon.identity.auth.otp.abstractotp",
+        immediate = true
+)
+public class AuthenticatorServiceComponent {
+
+    private static final Log log = LogFactory.getLog(AuthenticatorServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        try {
+            log.debug("OTP authenticator is activated");
```

Comment:
Does this log line provide any value?

Reply:
Removed the activate method since this does not register any service.